src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferConstCharOverConstUnitStringAnalyzer.cs

```diff
@@ -0,0 +1,105 @@
+                                                                                      DiagnosticCategory.Performance,
+                                                                                      RuleLevel.IdeSuggestion,
+                                                                                      s_localizableDescription,
+                                                                                      isPortedFxCopRule: false,
+                                                                                      isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterCompilationStartAction(compilationContext =>
+            {
+                // Check that the object is a StringBuilder
+                if (!compilationContext.Compilation.TryGetOrCreateTypeByMetadataName(WellKnownTypeNames.SystemTextStringBuilder, out INamedTypeSymbol? stringBuilderType))
+                {
+                    return;
+                }
+
+                IMethodSymbol appendStringMethod = stringBuilderType
+                    .GetMembers("Append")
+                    .OfType<IMethodSymbol>()
+                    .WhereAsArray(s =>
+                        s.Parameters.Length == 1 &&
+                        s.Parameters[0].Type.SpecialType == SpecialType.System_String)
+                    .FirstOrDefault();
+                if (appendStringMethod is null)
+                {
+                    return;
+                }
+
+                compilationContext.RegisterOperationAction(operationContext =>
+                {
+                    var invocationOperation = (IInvocationOperation)operationContext.Operation;
+                    if (invocationOperation.Arguments.Length < 1)
+                    {
+                        return;
+                    }
+
+
+                    if (!invocationOperation.TargetMethod.Equals(appendStringMethod))
+                    {
+                        return;
+                    }
+
+                    ImmutableArray<IArgumentOperation> arguments = invocationOperation.Arguments;
+                    IArgumentOperation firstArgument = arguments[0];
+
+                    // We don't handle class fields. Only local declarations
+                    if (!IsConstVar(firstArgument))
+                    {
+                        return;
+                    }
+
+                    if (firstArgument.Value is ILocalReferenceOperation argumentValue)
+                    {
+                        operationContext.ReportDiagnostic(argumentValue.CreateDiagnostic(Rule));
```

Comment:
Did we decide to not report this diagnostic on the local declaration? Looks fine to me, but just confirming since your earlier implementation did so. If we are not special casing the diagnostic location for locals, then I don't think you need to have the checks for `ILocalReferenceOperation` or `ILiteralOperation`. You can just write this as:

```csharp
                    if (IsConstVar(firstArgument))
                    {
                       operationContext.ReportDiagnostic(firstArgument.Value.CreateDiagnostic(Rule));
                    }
```

Reply:
I would even consider just inlining the conditional expression from `IsConstVar` directly here and remove `IsConstVar` completely.

Reply:
For the code fix, it might be fine to handle just the cases of `ILocalReferenceOperation` and `ILiteralOperation`, and let users manually fix any other case. For the analyzer, we should flag all operations with constant value that satisfy the value check.

Reply:
Done.